### ISSUE TITLE
Feature Suggestion - Open documents in client application and in same browser window

### ISSUE THREAD

**Issue by Tetronic** [nbelyh/VisioOnlineSpfxWebPart#22]
Hi, can you add two configuration properties as boolean types in your WebPart?

1. A boolean to control whether link targets from Visio files open in the browser or the default client application (for Excel, Word, PowerPoint and PDF files).

2. A boolean to control whether a click on a link target in the Visio file opens in a new browser tab or within the current active browser tab.

Thanks
Tetronic

**Comment by Tetronic** [comment]
Hi again,
did you maybe find the time to check by both feature suggestions (see this issue)?

Thanks
Tetronic

**Comment by nbelyh** [comment]
@Tetronic 
Yes, added it today. Thank you for reminding me 😅. Please check the 1.3 build
https://github.com/nbelyh/diagram-frame/releases/tag/1.3

<img width="644" alt="Image" src="https://github.com/user-attachments/assets/32acbbde-9a30-4cf2-9401-fd7685453e49" />

Other than that, I'm moving over to "native" Svg with SvgPublish webpart (no embedded Visio, but Visio export is supported)
https://github.com/nbelyh/svgpublishwebPart

The test page:
https://svgpublish-sb.unmanagedvisio.com/

**Comment by Tetronic** [comment]
@nbelyh : Thats awesome! Thanks a lot for the update! I will test this probably in around two weeks and give you feedback then.

**Comment by Tetronic** [comment]
@nbelyh Finally, I got the time to test your version:

The new settings are helpful and "Open Hyperlinks in New Window" works like a charm. 

However, I got two additional questions regarding the option "Force Open Office Files":

1. The option "Force Open Office Files" works for MS Office, but not for pdf files Could you add this functionality also for .pdf files?

2. If I set "Force Open Office Files" to false, the files are downloading on click as you wrote. My original wish was, instead of downloading, the setting should open the files in the client application, tough. E.g. if you have an office document in a SharePoint library, you always have the two options "open in browser" and "open in client app". So, my wish was meant to work as these two open functions:

"Force Open Office Files" = false => the office (and pdf) files open in the client application
"Force Open Office Files" = true => the office (and pdf) files open in the browser

Would that be possible?

**Comment by Tetronic** [comment]
Hi @nbelyh ,

did you maybe find the time to check my two questions from last month (see above)?

Thanks
Tetronic

**Comment by nbelyh** [comment]
@Tetronic 
Sorry, I have missed the question somehow 😅

1. Yes, including options for PDF is possible, but it depends on the browser. I mean, don't PDF files open alreayd like this? Do you use Office Web Apps to open them? 
2.  I can try to implement this using so-called "office file protocol" when you have it like this (supposed to force opening in a client, but not 100% sure - need to check):
```
   ms-word:ofe|u|https://your-sharepoint-url/file.docx
   ms-excel:ofe|u|https://your-sharepoint-url/file.xlsx
```

**Comment by Tetronic** [comment]
@nbelyh 
I checked again the opening, also compared to the opening in SharePoint:

1. For PDF: Yes, I use Office Web Apps in my SharePoint site to open the files. I tested both cases in SharePoint standard: open the pdf file whereas the SharePoint setting was set to "open in browser" and also when the setting was "open in client app". In both cases the pdf files was opened with Office Web Apps as shown in the following screenshot:

![Image](https://github.com/user-attachments/assets/cdb3047f-0433-46f4-9db6-b4d0eeee99f7)

When I use your SPFx Webpart, no matter whether I set "Force Open Office Files" to false or to true, the pdf file was opened always in the browser:

![Image](https://github.com/user-attachments/assets/47b6b643-19ae-43e0-a216-1e453c33b350)

I tested that with three browsers (Firefox, Chrome and Edge) and they behave all the same.
So, the only question is, can you change your current link to pdf files, so that the pdf files would be opened with Office Web Apps? That would be great.


2. Regarding office files: I tested that also again in the SharePoint standard way: 

SharePoint setting = "open in browser":  Office files are opened with Office Web Apps
SharePoint setting = "open in client app": I found again two different cases:
 a) the office client application is installed on my computer: The office file opens in the client app.
 b) the office client application is NOT installed on my computer: nothing happens if I click to open the file. 

So, the file won't be downloaded in the SharePoint standard way in either case, in contrast to your SPFx Webpart.

It would be great, if you could check making the link with the office file protocol as you suggested, so your webpart works like the SharePoint standard way.

**Comment by Tetronic** [comment]
Hi @nbelyh,

did you maybe find the time to check my two questions from September (see above)?

Thanks
Tetronic

**Comment by Tetronic** [comment]
Hello @nbelyh

my two little questions from above seem to be still open. Could you check if you can solve or answer them, briefly?

Thanks
Tetronic

**Comment by Tetronic** [comment]
@nbelyh : Update from my side:
I took your latest code and updated it in my environment to include both functions from above.

Now: 
- pdf files are also opened in office web apps, if "Force Open Office Files" == true
- pdf and other office files open in client application if "Force Open Office Files" == false

If you want, I can do a Pull Request, just drop me a brief comment here.
